### PR TITLE
 Use kube-batch as scheduler by default when gang-scheduling is enabled

### PR DESCRIPTION
**Pull request by jq**
Use kube-batch as scheduler by default when gang-scheduling is enabled, if scheduler is set, log warning message.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/100)

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -124,6 +124,32 @@
   revision = "1624edc4454b8682399def8740d46db5e4362ba4"
   version = "v1.1.5"
 
+[[projects]]
+  branch = "master"
+  digest = "1:d6ec8799b1f86397187016bfc7e129e51726e9b46106b0ca52f5478f4279a4e8"
+  name = "github.com/kubeflow/mpi-operator"
```
Comment (rongou) [inline Gopkg.lock]: This doesn't look right. It shouldn't depend on itself.

Comment (terrytangyuan) [reply Gopkg.lock]: @jq I think you need to do a `dep ensure` again and push your new changes in `Gopkg.lock` which should resolve this issue.

Comment (jq) [reply Gopkg.lock]: remove it directly

**Comment by rongou** [comment]
Ah I think the `solve-meta` section is a new feature in `dep`.

/lgtm
/approve

**Pull request merged**